amplitude/storages/inmemory_event_storage.go

```diff
@@ -1,34 +1,138 @@
 package storages
 
 import (
+	"sort"
+	"sync"
+	"time"
+
 	"github.com/amplitude/analytics-go/amplitude/types"
 )
 
-func NewInMemoryEventStorage(capacity int) types.EventStorage {
+func NewInMemoryEventStorage(chunkSize, sizeDivider int) types.EventStorage {
+	if sizeDivider < 1 {
+		sizeDivider = 1
+	}
+
+	currentChunkSize := chunkSize / sizeDivider
+	if currentChunkSize < 1 {
+		currentChunkSize = 1
+	}
+
 	return &inMemoryEventStorage{
-		capacity: capacity,
-		events:   make([]*types.Event, 0, capacity),
+		chunkSize:        chunkSize,
+		currentChunkSize: currentChunkSize,
+		sizeDivider:      sizeDivider,
```

Comment:
I think the chuck size would be better maintained by the AmplitudeResponseProcessor. It is possible multiple "clients" could share the same storage, with different chunk sizing/logic.